src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/WriteSPParksSites.cpp

```diff
@@ -0,0 +1,146 @@
+#include "WriteSPParksSites.hpp"
+
+#include "simplnx/Common/Constants.hpp"
+#include "simplnx/DataStructure/DataArray.hpp"
+#include "simplnx/DataStructure/Geometry/ImageGeom.hpp"
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/Utilities/FilterUtilities.hpp"
+
+#include <fstream>
+
+namespace fs = std::filesystem;
+using namespace nx::core;
+
+namespace
+{
+
+Result<> WriteHeader(const DataStructure& dataStructure, const WriteSPParksSitesInputValues* inputValues, std::ofstream& outfile)
+{
+  SizeVec3 dims = dataStructure.getDataAs<ImageGeom>(inputValues->ImageGeomPath)->getDimensions();
+  auto& featureIds = dataStructure.getDataAs<Int32Array>(inputValues->FeatureIdsArrayPath)->getDataStoreRef();
+
+  const size_t totalPoints = featureIds.getNumberOfTuples();
+
+  outfile << "-"
+          << "\n";
+  outfile << "3 dimension"
+          << "\n";
+  outfile << totalPoints << " sites"
+          << "\n";
+  outfile << "26 max neighbors"
+          << "\n";
+  outfile << "0 " << dims[0] << " xlo xhi"
+          << "\n";
+  outfile << "0 " << dims[1] << " ylo yhi"
+          << "\n";
+  outfile << "0 " << dims[2] << " zlo zhi"
+          << "\n";
+  outfile << "\n";
+  outfile << "Values"
+          << "\n";
+  outfile << "\n";
+
+  return {};
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+Result<> WriteFile(const DataStructure& dataStructure, const WriteSPParksSitesInputValues* inputValues, std::ofstream& outfile, const IFilter::MessageHandler& messageHandler,
+                   const std::atomic_bool& shouldCancel)
+{
+  SizeVec3 dims = dataStructure.getDataAs<ImageGeom>(inputValues->ImageGeomPath)->getDimensions();
+  auto& featureIds = dataStructure.getDataAs<Int32Array>(inputValues->FeatureIdsArrayPath)->getDataStoreRef();
+
+  size_t totalpoints = featureIds.getNumberOfTuples();
+
+  auto start = std::chrono::steady_clock::now();
+
+  // Buffer the output with 4096 Bytes which is typically the size of a "Block" on a
+  // modern Hard Drive. This should speed up the writes considerably
+  // char buffer[4096];
+  // outfile.rdbuf()->pubsetbuf(buffer, 4096);
```

Comment:
```suggestion
```
If you want you can leave a comment about how streams automatically handle buffer sizing now, but I felt it might be redundant so I opted for complete removal